### PR TITLE
Added AI Flags Documentation To Directory For New Documentation Site

### PR DESCRIPTION
**Pull request by anrichtait**
## **Description**
This pull request adds the file `ai_flags.md` to the `/docs` directory for the new documentation website.

## **Future contributions**
I look forward to contributing to more documentation. Please let me know if this is the correct directory to add documentation to in the future.

## **Discord Contact Info**
toonv#26526

### CODE REVIEW

File: docs/ai_flags.md
Comment (AlexOn1ine) [inline docs/ai_flags.md:36]: Instead of listing bullet points (which is just a subset of the functionality) I would summarize because the general gist of this flag is to prevent AI using moves that will fail, not nessasaty poor. How poor a move is is usually checked in `AI_FLAG_CHECK_VIABILITY`

File: docs/ai_flags.md
Comment (AlexOn1ine) [inline docs/ai_flags.md:58]: This is a weaker form of `AI_FLAG_TRY_TO_FAINT` which I would add as an additional comment.

**Review by AlexOn1ine — CHANGES_REQUESTED** [review]
Please add somewhere at the top that generally it is recommended to use `AI_FLAG_CHECK_BAD_MOVE`, `AI_FLAG_CHECK_VIABILITY`, `AI_FLAG_TRY_TO_FAINT` for good AI. Everything else should be used with consideration

**Comment by AlexOn1ine** [comment]
Thanks for the changes. I'll wait for @Pawkkie for feedback before merging

**Comment by Pawkkie** [comment]
> Thanks for the changes. I'll wait for @Pawkkie for feedback before merging

This looks good to me, I'll backport the changes to the wiki page once this gets merged.

My only piece of feedback is that I'm not sure "Keep in mind that this is a weaker form of `AI_FLAG_TRY_TO_FAINT`." is accurate when talking about AI_FLAG_STRONGEST_MOVE. Try To Faint has a more detailed clause for handling OHKOs, but doesn't handle 2HKOs at all. Strongest Move doesn't have as intelligent handling of OHKOs (it doesn't consider who's attacking first), but DOES handle 2HKOs.

If we want a comparative clause adding to that section, I might suggest something like "Keep in mind that this is weaker than AI_FLAG_TRY_TO_FAINT at scoring OHKOs" instead. @AlexOn1ine thoughts?

**Comment by AlexOn1ine** [comment]
> > Thanks for the changes. I'll wait for @Pawkkie for feedback before merging
> 
> This looks good to me, I'll backport the changes to the wiki page once this gets merged.
> 
> My only piece of feedback is that I'm not sure "Keep in mind that this is a weaker form of `AI_FLAG_TRY_TO_FAINT`." is accurate when talking about AI_FLAG_STRONGEST_MOVE. Try To Faint has a more detailed clause for handling OHKOs, but doesn't handle 2HKOs at all. Strongest Move doesn't have as intelligent handling of OHKOs (it doesn't consider who's attacking first), but DOES handle 2HKOs.
> 
> If we want a comparative clause adding to that section, I might suggest something like "Keep in mind that this is weaker than AI_FLAG_TRY_TO_FAINT at scoring OHKOs" instead. @AlexOn1ine thoughts?

Yeah, I like this suggestion

**Comment by anrichtait** [comment]
> > > Thanks for the changes. I'll wait for @Pawkkie for feedback before merging
> > 
> > This looks good to me, I'll backport the changes to the wiki page once this gets merged.
> > 
> > My only piece of feedback is that I'm not sure "Keep in mind that this is a weaker form of `AI_FLAG_TRY_TO_FAINT`." is accurate when talking about AI_FLAG_STRONGEST_MOVE. Try To Faint has a more detailed clause for handling OHKOs, but doesn't handle 2HKOs at all. Strongest Move doesn't have as intelligent handling of OHKOs (it doesn't consider who's attacking first), but DOES handle 2HKOs.
> > 
> > If we want a comparative clause adding to that section, I might suggest something like "Keep in mind that this is weaker than AI_FLAG_TRY_TO_FAINT at scoring OHKOs" instead. @AlexOn1ine thoughts?
> 
> Yeah, I like this suggestion


No problem I'll make the changes in the morning and also setup the new docs site on my local machine and create a new PR to add it there as well.

**Review by AsparagusEduardo — APPROVED** [review]

**Review by pkLucid — APPROVED** [review]

**Review by AlexOn1ine — APPROVED** [review]

**Pull request merged**
